Source/Extensions/Blazorise.DataGrid/DataGridColumn.cs

```diff
@@ -39,26 +41,39 @@ public DataGridColumn()
         valueGetter = new( () => FunctionCompiler.CreateValueGetter<TItem>( Field ) );
         valueSetter = new( () => FunctionCompiler.CreateValueSetter<TItem>( Field ) );
         sortFieldGetter = new( () => FunctionCompiler.CreateValueGetter<TItem>( SortField ) );
+
+        currentSortDirection[DataGridSortMode.Single] = SortDirection;
+        currentSortDirection[DataGridSortMode.Multiple] = SortDirection;
+        currentFilterMethod = FilterMethod;
+
+        Filter?.Subscribe( OnSearchValueChanged );
     }
 
     #endregion
 
     #region Methods
 
-    protected override void OnInitialized()
+    /// <summary>
+    /// Provides a way to generate column specific dependencies outside the Blazor engine.
+    /// </summary>
+    /// <param name="parentDataGrid"></param>
+    /// <param name="serviceProvider"></param>
+    internal void InitializeGeneratedColumn( DataGrid<TItem> parentDataGrid, IServiceProvider serviceProvider )
     {
-        base.OnInitialized();
+        this.ParentDataGrid = parentDataGrid;

```

Comment:
I guess this is the reason for `SuppressMessage` settings? Can we make it on a method level instead of making it a global setting? Easier to see it in the future if we directly set properties somewhere else.

Reply:
I like the supression file alot. It keeps all the supression in a single file and easily referable to with justification. You should not need to be "poluting" code with it.

There is no issue with the line you are linking. It is being set in the component itself.

The reason is for the setters on `Parameters` when newing up a Blazor component in the `Datagrid.AutoGenerateColumns`.
![image](https://github.com/Megabit/Blazorise/assets/22283161/eb91893c-55df-4277-a3af-040ed83d53b8)

But let me know if you still prefer to inline it.

Reply:
OK, we can leave at as it is.